linera-core/src/client/mod.rs

```diff
@@ -1281,7 +1277,12 @@ where
         let local_response = self.local_query(ChainInfoQuery::new(chain_id)).await?;
         let local_next = local_response.info.next_block_height;
 
-        tracker.push(chain_id, local_next.saturating_sub(1.into()));
+        // Record highest block height known locally for this chain.
+        if let Ok(highest_seen) = local_next.try_sub_one() {
+            tracker.push(chain_id, highest_seen);
+        }
+
+        // Find the first block in the batch that is higher than the highest block known locally.
```

Comment:
```suggestion
        /// Finds the first block in the batch that is higher than the highest block known locally.
```

Reply:
I thought we want these `///` comments on public methods only.

Reply:
Sorry! :facepalm: